lib/src/oauth2.dart

```diff
@@ -17,6 +18,11 @@ import 'log.dart' as log;
 import 'system_cache.dart';
 import 'utils.dart';
 
+/// The global HTTP client with basic retries. Used instead of retryForHttp for
+/// OAuth calls because the OAuth2 package requires a client to be passed. While
+/// the retry logic is more basic, this is fine for the publishing process.
+final _retryHttpClient = RetryClient(globalHttpClient);
```

Comment:
Do we want to pass any config to `RetryClient`?

Reply:
`when` and `whenError` might be worth defining, the defaults are a little lax:
https://github.com/dart-lang/http/blob/master/pkgs/http/lib/retry.dart#L151-L153

Like `whenError: (e, st) => e is IoException` might be fine.
`when: (BaseResponse response) => response.statusCode >= 500` might also be nice...

This is less important.. but something like that.